src/test_wasm_simd.py

```diff
@@ -0,0 +1,127 @@
+    f.write('}\n')
+  else:
+    f.write('%s test(%s a) {\n' % (returntype, datatype))
+    f.write('  return emscripten_%s_%s(a);\n' % (datatype, operation))
+    f.write('}\n')
+  f.close()
+  return f.name
+
+def compile_test_file(filename, clang, include):
+  p = subprocess.Popen([clang, '-w', '-S', '-emit-llvm', filename,
+                        '-O3', '-o-', '-I', include],
+                       stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  (output, err) = p.communicate()
+  if output == '':
+    print ('================================================================')
+    print (err.strip())
+    print ('================================================================')
+    return (False, 'Compiler error (see above)')
+  else:
+    return (True, output)
+
+def check_test_output(output, opcode, arity):
+  param_pattern = r', '.join(r'<.+> %\w' for x in range(arity))
+  operand_pattern = r', '.join(r'%\w' for x in range(arity))
+  pattern = (
+    r'define <.+> @test\(' + param_pattern + r'\) local_unnamed_addr #\d \{\s+' +
+    r'entry:\s+' +
+    r'(%.+) = ([A-Za-z0-9]+) .+ ' + operand_pattern + r'\s+' +
+    r'ret <.+> \1\s+' +
+    r'\}')
+  m = re.search(pattern, output)
+  if m:
+    if m.group(2) == opcode:
+      return (True, None)
+    else:
+      return (False, 'Expected operation %s but found %s' % (opcode, m.group(2)))
+  else:
+    print ('================================================================')
+    m = re.search(r'define .+ \{[^}]+\}', output)
+    if m:
+      print m.group(0).strip()
+      err = 'Compiled function not in intrinsic form (see above)'
+    else:
+      print output.strip()
+      err = 'Compiled function not found (see above)'
+    print ('================================================================')
+    return (False, err)
+
+def test_wasm_simd_instruction(datatype, operation, arity, returntype, opcode, clang, include):
+  filename = create_test_file(datatype, operation, arity, returntype)
+  (passed, output) = compile_test_file(filename, clang, include)
+  if passed:
+    (passed, err) = check_test_output(output, opcode, arity)
+    if passed:
+      print ('%s %s: PASSED' % (datatype, operation))
+    else:
+      print ('%s %s: FAILED: %s' % (datatype, operation, err))
+  else:
+    print ('%s %s: FAILED: %s' % (datatype, operation, output))
+  os.remove(filename)
+  return passed
+
+def test_wasm_simd(clang, include):
+  passed = 0
+  failed_expectedly = 0
+  failed_unexpectedly = 0
+  for (operation, arity, datatype, returntype, opcode, expect_failure) in wasm_simd_test_cases:
+    if test_wasm_simd_instruction(datatype, operation, arity, returntype, opcode, clang, include):
+      passed += 1
+    elif expect_failure:
+      failed_expectedly += 1
+    else:
+      failed_unexpectedly += 1
+  print ('Passes: %d' % passed)
+  print ('Failures (expected): %d' % failed_expectedly)
+  print ('Failures (unexpected): %d' % failed_unexpectedly)
```

Comment:
Here it's printing the string-format expression `'str with %s' % 'format'`. Strictly speaking it doesn't need parens to do so. I believe python2 syntax-wise it does need the space, although it sure *looks* a lot like python3 function printing.

Also because it's just calling print, we can drop the `%` formatting entirely and just do `print 'Passes:', passed`